app/presenters/rejected_application_choice.rb

```diff
@@ -0,0 +1,119 @@
+        quality_reasons << reasons.quality_of_application_other_what_to_improve
+      else
+        quality_reasons << reasons.send("quality_of_application_#{reason}_what_to_improve")
+      end
+    end
+
+    reason_details('quality_of_application', application_quality)
+  end
+
+  def qualifications
+    return {} unless reasons.qualifications_y_n.eql?('Yes')
+
+    qualifications = reasons.qualifications_which_qualifications.inject([]) do |qualification_reasons, reason|
+      if reason == 'other'
+        qualification_reasons << reasons.qualifications_other_details
+      else
+        qualification_reasons << I18n.t("reasons_for_rejection.qualifications_which_qualifications.#{reason}")
+      end
+    end
+
+    reason_details('qualifications', qualifications)
+  end
+
+  def interview_performance
+    return {} unless reasons.performance_at_interview_y_n.eql?('Yes')
+
+    reason_details('interview_performance', [reasons.performance_at_interview_what_to_improve])
+  end
+
+  def full_course
+    return {} unless reasons.course_full_y_n.eql?('Yes')
+
+    reason_details('full_course')
+  end
+
+  def offered_other_course
+    return {} unless reasons.offered_on_another_course_y_n.eql?('Yes')
+
+    reason_details('offered_on_another_course', [reasons.offered_on_another_course_details])
+  end
+
+  def honesty_and_professionalism_reasons
+    return {} unless reasons.honesty_and_professionalism_y_n.eql?('Yes')
+
+    honesty_and_professionalism = [reasons.honesty_and_professionalism_concerns_information_false_or_inaccurate_details,
+                                   reasons.honesty_and_professionalism_concerns_plagiarism_details,
+                                   reasons.honesty_and_professionalism_concerns_references_details,
+                                   reasons.honesty_and_professionalism_concerns_other_details].compact
+
+    reason_details('honesty_and_professionalism', honesty_and_professionalism)
+  end
+
+  def safeguarding_issues
+    return {} unless reasons.safeguarding_y_n.eql?('Yes')
+
+    safeguarding_issues = [reasons.safeguarding_concerns_candidate_disclosed_information_details,
+                           reasons.safeguarding_concerns_vetting_disclosed_information_details,
+                           reasons.safeguarding_concerns_other_details].compact
+
+    reason_details('safeguarding_issues', safeguarding_issues)
+  end
+
+  def additional_advice
+    return {} unless reasons.other_advice_or_feedback_y_n.eql?('Yes')
+
+    reason_details('additional_advice', [reasons.other_advice_or_feedback_details])
+  end
+
+  def interested_in_future_applications
+    return {} unless reasons.interested_in_future_applications_y_n.eql?('Yes')
+
+    reason_details(
+      'interested_in_future_applications',
+      [I18n.t('reasons_for_rejection.interested_in_future_applications.reason',
+              provider_name: course_option.course.provider.name)],
```

Comment:
Not a blocker, this could be abbreviated to `provider.name` because https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/models/application_choice.rb#L11

Reply:
I tried that but I could not get `build_stubbed` object to work in the specs and then noticed a comment about it (here https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/mailers/candidate_mailer.rb#L64).

Any suggestions on how I can work around that?

Reply:
No, you're right, it becomes a headache to stub, just ran into the same problem on the API changes.